lxd/auth/oidc/oidc.go

```diff
@@ -152,6 +153,50 @@ func (o *Verifier) Auth(ctx context.Context, w http.ResponseWriter, r *http.Requ
 	return claims.Subject, nil
 }
 
+// authenticateIDToken verifies the identity token and returns the ID token subject. If no identity token is given (or
+// verification fails) it will attempt to refresh the ID token.
+func (o *Verifier) authenticateIDToken(ctx context.Context, w http.ResponseWriter, idToken string, refreshToken string) (string, error) {
```

Comment:
As far as am aware there aren't any cases like this. Is that correct @nsklikas?

Reply:
I don't think that there are such cases. 

Only case that I can think of is if the ID Token provided is not there or if it is malformed (not properly signed, issued for another client, etc), then you shouldn't try to refresh the token as the refresh token would also be invalid and you can avoid doing that extra http call (perhaps this could be a possible attack?). Having said that I think that it is fine to try to refresh the token every time the id token verification fails.